### PR TITLE
Ticket2887 display raw field readings

### PR DESCRIPTION
**Pull request by esouthren**
### Description of work

Unit value PVs have been added to the IOC to allow user scientists to log values without them varying if they switch units. 
For example, when logging the raw device reading, if the user changes units the value of the readback will spike dramatically as it changes from Tesla to Gauss.

lksh_raw

By adding FIELD:TESLA and FIELD:GAUSS PVs, we can maintain the overall log of the readback regardless of which unit we are in. Below, the units are changed from Tesla to Gauss but the RAW:GAUSS value remains unchanged. 
lsh_gauss

### To test

[Ticket 2887](https://github.com/ISISComputingGroup/IBEX/issues/2887)

### Acceptance criteria

See also: [OPI pull request](https://github.com/ISISComputingGroup/ibex_gui/pull/651)
See also: [Test Files pull request](https://github.com/ISISComputingGroup/EPICS-IOC_Test_Framework/pull/62)
---

#### Code Review

- [ ] **Pertitent information and a copy of the manual has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)**
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Comment by esouthren** [comment]
Added pull request for DBUnitChecker (added 'G' to allow tests to pass)

https://github.com/ISISComputingGroup/EPICS-DbUnitChecker/compare/Ticket2887_addition_to_db_checker?expand=1

File: LKSH460/LKSH460-IOC-01App/Db/lakeshore460_channels.template [hunk 1]
```diff
@@ -1,26 +1,52 @@
...
+	field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:UNIT:SP")
 }
 
 record(ai, "$(P)$(CHANNEL):MAXREADING")
 {
 	field(DESC, "Max reading")
 	field(DTYP, "stream")
-	field(SCAN, "2 second")
+	#field(SCAN, "2 second")
```
[the remark this answers is not in the document]
Comment (esouthren) [reply LKSH460/LKSH460-IOC-01App/Db/lakeshore460_channels.template]: Removed

File: LKSH460/LKSH460-IOC-01App/Db/lakeshore460_channels.template [hunk 2]
```diff
@@ -232,10 +297,41 @@ record(stringout, "$(P)$(CHANNEL):RELMODESET:UNIT:CALC:STR")
...
+    field(SDIS, "$(P)$(CHANNEL):DISABLE")
+	info(INTEREST, "LOW")	
+	field(EGU, "")
+}
+
 record(bi, "$(P)$(CHANNEL):MODE") 
 {
 	field(DESC, "Get the field reading mode")
     field(DTYP, "stream")
+	#field(SCAN, "5 second")
```
[the remark this answers is not in the document]
Comment (esouthren) [reply LKSH460/LKSH460-IOC-01App/Db/lakeshore460_channels.template]: Removed

File: LKSH460/LKSH460-IOC-01App/Db/lakeshore460_channels.template [hunk 3]
```diff
@@ -232,10 +297,41 @@ record(stringout, "$(P)$(CHANNEL):RELMODESET:UNIT:CALC:STR")
...
+	field(DTYP, "stream")
+	field(INP, "@devLakeshore_460.proto getRelModeSetpoint($(CHANNEL)) $(PORT)")
+	info(INTEREST, "HIGH")	
+	info(ARCHIVE, "VAL")
+	field(EGU, "")
+	field(PREC, "4")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:$(CHANNEL):RELMODESET")
+    field(SDIS, "$(P)DISABLE")
+	#field(FLNK, "$(P)$(CHANNEL):UNIT:MULT")
```
[the remark this answers is not in the document]
Comment (esouthren) [reply LKSH460/LKSH460-IOC-01App/Db/lakeshore460_channels.template]: Removed

**Comment by Tom-Willemsen** [comment]
Still getting failures in DbUnitChecker due to duplicated PVs.

```
Beginning PV unit tests...

Running tests...
----------------------------------------------------------------------
................F...................
======================================================================
ERROR [0.001s]: test_multiple_pvs_warning (__main__.TestPVUnits)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Instrument\Apps\EPICS\ISIS\DbUnitChecker\master\db_checker.py", line 54, in wrapper
    func(self, *args, **kwargs)
  File "C:\Instrument\Apps\EPICS\ISIS\DbUnitChecker\master\db_checker.py", line 54, in wrapper
    func(self, *args, **kwargs)
  File "C:\Instrument\Apps\EPICS\ISIS\DbUnitChecker\master\db_checker.py", line 54, in wrapper
    func(self, *args, **kwargs)
  File "C:\Instrument\Apps\EPICS\ISIS\DbUnitChecker\master\db_checker.py", line 100, in test_multiple_pvs_warning
    "Multiple fields on PVs in {}".format(self.db.directory), failures))
AssertionError: Multiple fields on PVs in C:\Instrument\Apps\EPICS\ioc\master\LKSH460\db\lakeshore460_channels.db
   -> Multiple instances of $(P)UNIT:SP
   -> Multiple instances of $(P)UNIT

----------------------------------------------------------------------
Ran 36 tests in 0.008s

FAILED (errors=1)
```

**Comment by esouthren** [comment]
'''
C:\Instrument\Apps\EPICS\ioc\master\LKSH460\db>python C:\Instrument\Apps\EPICS\ISIS\DbUnitChecker\master\db_checker.py -i .
Scanning files...
Beginning PV unit tests...

Running tests...
----------------------------------------------------------------------
..................
----------------------------------------------------------------------
Ran 18 tests in 0.005s

OK

Generating XML reports...
PV unit tests complete (Took 0.235 sec)

C:\Instrument\Apps\EPICS\ioc\master\LKSH460\db>
'''

Passing now!

**Pull request merged**